diffkemp/simpll/DifferentialFunctionComparator.h

```diff
@@ -31,7 +31,10 @@ class DifferentialFunctionComparator : public FunctionComparator {
                                    const Function *F2,
                                    GlobalNumberState *GN,
                                    const DebugInfo *DI)
-            : FunctionComparator(F1, F2, GN), DI(DI) {};
+            : FunctionComparator(F1, F2, GN), DI(DI) {
+        LayoutL = new DataLayout(F1->getParent());
+        LayoutR = new DataLayout(F2->getParent());
```

Comment:
Why allocating new DataLayout when you can get it directly from the module (F1->getParent()->getDataLayout())? This code would cause a memory leak because you do not free the memory anywhere.